realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -458,6 +454,17 @@ public void onSuccess() {
                 looperThread.testComplete();
             }
         });
+
+        // Wait for all async tasks finish to ensure the async transaction posted callback will arrive first.
+        TestHelper.resetRealmThreadExecutor();
```

Comment:
I think when that happens, we should fix it. `NoSuchFieldException` and `IllegalAccessException` doesn't sounds like anything can be recovered...